.github/workflows/ci-pipeline.yml

```
@@ -50,13 +61,17 @@ jobs:
       # Approve the CI's PR if the 'lint' job succeeded
       # Approved by the 'github-actions' user; a PR can't be approved by its author
       - name: PR approval
-        if: ${{ needs.lint.result == 'success' }}
+        if: |
+          needs.lint.result == 'success' &&
+          needs.lint.outputs.allow_auto_merge == true
```

I am not sure about this condition, [outputs are strings](https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context), I think you should use:
```suggestion
          needs.lint.outputs.allow_auto_merge == 'true'
```